packages/core/src/components/SessionTimeout/helpers.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class Session extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      now: this.props.now(),
+      expiresAt: this.props.expiresAt
+    };
+  }
+
+  componentWillMount() {
+    this._interval = setInterval(
+      () => this.setState(() => ({ now: this.props.now() })),
```

Comment:
this is very similar to the timer code (for toast notifications) wonder if we can do some reuse..?

Reply:
SGTM. I'll work on it next week. Thanks for the review.